packages/purpose-process/test/getPurposes.test.ts

```diff
@@ -526,6 +532,135 @@ describe("getPurposes", async () => {
     expect(result.totalCount).toBe(0);
     expect(result.results).toEqual([]);
   });
+  it("should get the purposes if the requester is an e-service delegated consumer", async () => {
+    const delegation = getMockDelegation({
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: mockPurpose1.eserviceId,
+      delegatorId: mockPurpose1.consumerId,
+      delegateId: consumerId1,
+      state: delegationState.active,
+    });
```

Comment:
`delegationId` should be the one from the purpose and delegate should be different from the consumer... maybe you need to create a new mock purpose `delegatedPurpose` and do something like:

```suggestion
    const delegation = getMockDelegation({
      delegationId: delegatedPurpose.delegationId,
      kind: delegationKind.delegatedConsumer,
      eserviceId: delegatedPurpose.eserviceId,
      delegatorId: delegatedPurpose.consumerId,
      delegateId: generateId<DelegationId>(),
      state: delegationState.active,
    });
```